### DISCUSSION TITLE
FastLED backlight and Color Mapping modification

### DISCUSSION THREAD

**Discussion by rassu05** [tttapa/Control-Surface#609] [Q&A] [answered]
Hi!
I'm building a midi controller, and I've come so far that I have buttons with addressable LEDs underneath. 
The LED is activated through the software when the button is pressed. Now I want to add a backlight. I want all LEDs to glow a little blue when the midi value sent to them is 0.  
I hope one way is to change the color mapping. I would also like to change the color of the midi value 127. 
1. Is this possible and how? 
2. Is it possible to assign different color mapping to different groups of LEDs?

Thank you so much for creating a WONDERFUL library, I don't know what I would do without it!

**Comment by tttapa** [comment, marked as answer]
Does this example answer your question? https://tttapa.github.io/Control-Surface-doc/Doxygen/d9/d6d/10_8Note-FastLED-ColorMapper_8ino-example.html

You can use the color mapper to convert the MIDI velocity to any FastLED color value you want.

**Reply by rassu05** [reply]
It's a bit useful, but I don't quite understand how to do it yet. I understand that there is no separate guide for this? Could you give another example?
For example:
MIDI velocity 0 = half power blue
MIDI velocity 126 = Red
MIDI velocity 127 = Green

**Reply by tttapa** [reply]
You can just encode those rules as a switch statement:
```cpp
// Create a functor that maps the velocity and the index of a note to a color.
struct BacklightColorMapper {
  CRGB operator()(uint8_t velocity, uint8_t index) const {
    (void) index; // unused
    switch (velocity) {
      case 0:   return CRGB(0,   0,   127);
      case 126: return CRGB(255, 0,   0);
      case 127: return CRGB(0,   255, 0);
      default:  return CRGB(0,   0,   0);
    }
  }
};
 
NoteRangeFastLED<leds.length, BacklightColorMapper> midiled {
  leds,
  MIDI_Notes::C(4),
};
```
